### PR TITLE
applications: nrf_desktop: Add known issues section

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-976

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
The plan is to link this section in https://github.com/nrfconnect/sdk-nrf/wiki/Known-issues

**Comment by greg-fer** [comment]
I'm not sure this Known issues section should be added in the conceptual documentation. I'm personally against it. Known issues are usually added in Release notes. There are several reasons for this:
- Known issues are "issues", so bugs. Bugs are not the intended way of working, unlike conceptual documentation.
- Known issues can be fixed in short or long term, which means that adding them to very static conceptual docs will create the need to update the conceptual docs on regular basis, even if nothing else changes in the conceptual docs

I'm not up-to-date about the Known issues section in the Wiki on GitHub, but I think the Known issues list should go there, and not into the conceptual documentation. We can add a note with a link to nRF Desktop introductory section, but not more. @ru-fu , what do you think?

**Comment by ru-fu** [comment]
Known issues for a release should go into the release notes. Known issues for a release that are discovered **after** the release is tagged (and that therefore cannot go into the release notes anymore) should go on the wiki page.
So linking to this list from the wiki does not make sense, because the wiki page is for the latest release, and that release doesn't have this list yet.

In the conceptual documentation, we could list limitations - thus things that we know don't work but don't plan to fix. But having known issues for a development version seems strange, because if we know about them, we're planning to fix them, or aren't we?

That being said, there are also discussions about Known issues pages in nrfxlib - but the situation there is a bit different, since we have (regular) binary releases there, and each of these releases might have known issues.

**Comment by pdunaj** [comment]
Hi @greg-fer , @ru-fu , the idea behind putting the known issues in the application documentation is to allow customers to have easier access to it. The assumption we have is that customers will look directly into the application documentation.

btw. In my previous company there was very specific way of documenting errata for SW components, where brief information was stated. Do we have anything like that standardized here?

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -1521,6 +1521,19 @@ These are valid for events that have many listeners or sources, and are shared f
...
 
+.. _nrf_desktop_known_issues:
+
+Known issues
+************
+
+* Boot reports are not sent
+* Dongle - Keyboard connection deteriorates badly with split_ll
+* Unable to re-bond keyboard with OSX host
+* Dongle unable to setup connection with Desktop peripherals
```
Comment (jhn-nordic) [inline applications/nrf_desktop/README.rst:1532]: Is it always, frequently, or a corner case?

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -1521,6 +1521,19 @@ These are valid for events that have many listeners or sources, and are shared f
...
+* Bond switching in Gaming Mouse stops working
```
Comment (jhn-nordic) [inline applications/nrf_desktop/README.rst:1533]: Is it always, frequently, or a corner case?

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -1521,6 +1521,19 @@ These are valid for events that have many listeners or sources, and are shared f
...
    doc/event_rel_modules.rst
 
+.. _nrf_desktop_known_issues:
+
+Known issues
+************
+
+* Boot reports are not sent
+* Dongle - Keyboard connection deteriorates badly with split_ll
+* Unable to re-bond keyboard with OSX host
```
Comment (jhn-nordic) [inline applications/nrf_desktop/README.rst:1531]: OS X is now called macOS (since macOS 10.12: Sierra from 2016)

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -1521,6 +1521,19 @@ These are valid for events that have many listeners or sources, and are shared f
...
+* Dongle unable to setup connection with Desktop peripherals
+* Bond switching in Gaming Mouse stops working
+* Config channel should use write without response
```
Comment (jhn-nordic) [inline applications/nrf_desktop/README.rst:1534]: Is this an issue or a planned improvement?
Is the issue that the config channel is slow, that it affects the report rate or both.

**Comment by greg-fer** [comment]
Hey @pdunaj , to follow up on the "where to place known issues".
I'd put it this way:
- If known issues is something you as a team are planning to fix, the application documentation is not the place to have them, and they should be place in the wiki. Besides, they will have to go into the wiki regardless, and we don't want to edit them in two places.
- If (some or all) known issues are to stay with us and won't be fixed, they should be integrated into the documentation -- as notes or by rewriting documentation sections, because they affect the way the application is working.

This is the ideal approach for known issues, not only for Nordic. Some companies might do it otherwise, but it is always a "lesser-evil" compromise.

**Comment by greg-fer** [comment]
Had an additional talk with @MonikaLieLarsen about this, and here's her take:

> Known issues is what we put in as an afterthought, in the release notes, after everyting else is finalized and it's too late to fix. And this is where customers expect to find it. 
> If developers expect to have known issues for a component, they can point to the Release Notes from the conceptual documentation, as in "See Release Notes for known issues".

**Comment by greg-fer** [comment]
Discussed this f2f with @pdunaj . There will be a note with a link to the wiki in the nRF Desktop docs, and known issues will be listed in the wiki.

Pawel also mentioned some potential improvements to the known issue documentation process. We can discuss these during a separate meeting, @ru-fu .

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -727,6 +727,9 @@ The nRF Desktop application is built the same way to any other |NCS| application
 
 .. include:: /includes/build_and_run.txt
 
+.. note::
+    The information about the known issues can be found in :ref:`release_notes` and in the `Known issues`_ wiki page
```
Comment (greg-fer) [inline applications/nrf_desktop/README.rst:731]: I would rephrase it slightly:
```
Information about the known issues in nRF Desktop can be found in |NCS|'s :ref:`release_notes` and on the `Known issues`_ wiki page.
```
